core/src/main/java/org/elasticsearch/index/query/DisMaxQueryParser.java

```diff
@@ -100,16 +98,15 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
             throw new QueryParsingException(parseContext, "[dis_max] requires 'queries' field");
         }
 
-        if (queries.isEmpty()) {
-            return null;
+        DisMaxQueryBuilder disMaxQuery = new DisMaxQueryBuilder();
+        disMaxQuery.tieBreaker(tieBreaker);
+        disMaxQuery.queryName(queryName);
+        disMaxQuery.boost(boost);
+        for (QueryBuilder query : queries) {
+            disMaxQuery.add(query);
         }
-
-        DisjunctionMaxQuery query = new DisjunctionMaxQuery(queries, tieBreaker);
-        query.setBoost(boost);
-        if (queryName != null) {
-            parseContext.addNamedQuery(queryName, query);
-        }
-        return query;
+        disMaxQuery.validate();
```

Comment:
sure, missed that one. I also deleted it from BoolQueryParser I think. Will do so if when I spot it in other places except tests.